scopes/angular/angular/angular.env.ts

```diff
@@ -1,65 +1,117 @@
-    //      */
-    //     private compiler: CompilerMain,
-    //
-    //     /**
-    //      * webpack extension.
-    //      */
-    //     private webpack: WebpackMain,
-    //
-    //     /**
-    //      * workspace extension.
-    //      */
-    //     private workspace: Workspace,
-    //
-    //     /**
-    //      * pkg extension.
-    //      */
-    //     private pkg: PkgMain,
-    //
-    //     /**
-    //      * tester extension
-    //      */
-    //     private tester: TesterMain,
-    //
-    //     private config: ReactMainConfig,
-
-    private eslint: ESLintMain
+    private jestAspect: JestMain,
+    private tsAspect: TypescriptMain,
+    private compiler: CompilerMain,
+    private webpack: WebpackMain,
+    private workspace: Workspace,
+    private pkg: PkgMain,
+    private tester: TesterMain,
+    private eslint: ESLintMain,
+    private ngPackagrAspect: NgPackagrMain
   ) {}
 
-  async __getDescriptor() {
+  /**
+   * Returns the Environment descriptor
+   * Required for any task
+   */
+  async __getDescriptor(): Promise<EnvDescriptor> {
     return {
       type: 'angular',
     };
   }
 
+  private createNgPackgrCompiler(compilerOptions: Partial<CompilerOptions> = {}) {
+    return this.ngPackagrAspect.createCompiler({
+      ...compilerOptions,
+    });
+  }
+
+  getCompiler(compilerOptions: Partial<CompilerOptions> = {}) {
+    return this.createNgPackgrCompiler(compilerOptions);
+  }
+
+  private getCompilerTask() {
+    return this.compiler.createTask('NgPackagrCompiler', this.getCompiler());
+  }
+
+  /**
+   * Returns the component build pipeline
+   * Required for `bit build`
+   */
+  getBuildPipe(): BuildTask[] {
+    return [this.getCompilerTask()];
+  }
+
+  /**
+   * Returns a paths to a function which mounts a given component to DOM
+   * Required for `bit build`
+   */
+  getMounter() {
+    // return require.resolve('./mount');
+    return ''; // TODO(ocombe)
+  }
+
+  /**
+   * Returns a path to a docs template.
+   * Required for `bit build`
+   */
+  getDocsTemplate() {
+    // return require.resolve('./docs');
+    return ''; // TODO(ocombe)
+  }
+  /**
+   * Returns a bundler for the preview.
+   * Required for `bit build` & `build start`
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async getBundler(context: BundlerContext, transformers: any[]): Promise<Bundler> {
+    return null as any; // TODO(ocombe)
+  }
+
+  /**
+   * returns and configures the component linter.
+   * Required for `bit lint`
+   */
+  getLinter(): Linter {
+    return this.eslint.createLinter({
+      config: eslintConfig,
+      // resolve all plugins from the angular environment.
+      pluginPath: __dirname,
+    });
+  }
+
+  /**
+   * Returns the list of dependencies
+   * Required for any task
+   */
   getDependencies() {
     return {
       dependencies: {
         '@angular/common': '-',
         '@angular/core': '-',
-        tslib: '-',
+        tslib: '^2.0.0',
```

Comment:
is this required to be dependency of any component uses the angular env?

Reply:
Yes, otherwise ngPackagr will print a warning that it should be a dependency and not a peer dependency. Not 
sure why though